pyv/transpiler.py

```diff
@@ -0,0 +1,486 @@
+            if getattr(node, 'annotation', None):
+                typename = self._typename_from_annotation(node, attr="annotation")
+                signature.append(typename)
+        elif not is_void_function(node):
+            if getattr(node, 'returns', None):
+                typename = self._typename_from_annotation(node, attr="returns")
+                signature.append(typename)
+
+        return " ".join(signature)
+
+    def visit_FunctionDef(self, node):
+        body = "\n".join([self.indent(self.visit(n)) for n in node.body])
+        return f"{self.function_signature(node)} {{\n{body}\n}}"
+
+    def visit_Return(self, node):
+        if node.value:
+            ret = self.visit(node.value)
+            return f"return {ret}"
+        return "return"
+
+        if node.value:
+            return "return {0}".format(self.visit(node.value))
+        return "return"
+
+    def visit_Lambda(self, node):
+        raise AstNotImplementedError("Lambdas are not supported yet.", node)
+
+    def visit_Attribute(self, node):
+        attr = node.attr
+
+        value_id = self.visit(node.value)
+
+        if is_list(node.value):
+            if node.attr == "append":
+                return f"{value_id} <<"
+        if not value_id:
+            value_id = ""
+        ret = f"{value_id}.{attr}"
+        if ret in self._attr_dispatch_table:
+            return self._attr_dispatch_table[ret](self, node)
+        return value_id + "." + attr
+
+    def visit_range(self, node, vargs: List[str]) -> str:
+        if len(node.args) == 1:
+            return f"0..{vargs[0]}"
+        elif len(node.args) == 2:
+            return f"{vargs[0]}..{vargs[1]}"
+
+        raise Exception(
+            "encountered range() call with unknown parameters: range({})".format(vargs)
+        )
+
+    def visit_print(self, node, vargs: List[str]) -> str:
+        args = []
+        total_args = []
+        for arg in node.args:
+            if isinstance(arg, ast.Constant) and isinstance(arg.value, str):
+                args.append(arg.value.replace("'", ""))
+            elif get_inferred_v_type(arg) == "string":
+                args.append(f"${{{self.visit(arg)}}}")
+            else:
+                if args:
+                    total_args.append(f"'{' '.join(args)}'")
+                    args = []
+                total_args.append(f"({self.visit(arg)}).str()")
+        if args:
+            total_args.append(f"'{' '.join(args)}'")
+        total_args = " + ' ' + ".join(total_args)
+
+        return f"println({total_args})"
+
+    def visit_cast(self, node, vargs) -> Optional[str]:
+        typ = get_id(node.func)
+        if typ == 'bool':
+            if get_inferred_v_type(node.args[0]) in V_WIDTH_RANK:
+                return f'{self.visit(node.args[0])} != 0'
+            else:
+                return f'({self.visit(node.args[0])}).bool()'
+
+    def _dispatch(self, node, fname: str, vargs: List[str]) -> Optional[str]:
```

Comment:
As of https://github.com/adsharma/py2many/pull/352, all the other transpilers now use a generic `_dispatch` from the super class, which relies on bits of this function and `visit_print`/`visit_range`,etc being moved into `pyX/plugins.py`